### PR TITLE
Feature/66 generate apidoc

### LINKED ISSUE #54: Create initial Swagger json file for departments API endpoints
Use a tool to create the Swagger file for department REST endpoints. Swagger file is defined before creating the API endpoints (design first principle).

Definition of done
---
* [x] Swagger file exists in project.
* [x] Swagger file contains general description of API.
* [x] Swagger file contains specific details for each API endpoint.
* [x] Swagger file passes validation.
* [x] Swagger file is published and accessible via URL route.

### LINKED ISSUE #66: Auto-generate API documentation file
Refer to [Meteor SimpleREST issue #93](https://github.com/stubailo/meteor-rest/issues/93)

Definition of done
---
* [x] Review the issue 
* [x] Get an uderstanding of what code needs to be written
* [x] Fork or create a new package
* [x] Review Swagger spec for required fields for producing a Swagger file
* [x] Implement minimum required Swagger fields in JSON format
  * [x] When a user visits Swagger endpoint, they receive valid Swagger JSON
* [ ] Open PR on Meteor Simple REST

### PR DESCRIPTION
**Pull request by jykae**
Closes #66 
Closes #54 

# Proposed changes
- add package apinf:restivus-swagger
- add required swagger metadata for API, endpoints
- define route for generated swagger.json

@Digipalvelutehdas/dipor-team please review.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 .meteor/meteorite
 .idea
 .DS_Store
+packages
```
Comment (brylie) [inline .gitignore]: Remove this line.

**Comment by jykae** [comment]
Adding support for hostname, basePath, schemes & endpoint tags to group endpoints in Swagger UI.

**Comment by jykae** [comment]
Requested changes made, hostname basePath, schemes are dynamically supported by restivus-swagger package.

Disabled authentication for now and leaving it to another issue: https://github.com/Digipalvelutehdas/dipor-dashboard/issues/100

Basically calling GET endpoints without parameters work now with Swagger UI, I list improvements regarding Swagger UI support to new issue.

File: .gitignore (hunk 1 above)
Comment (brylie) [inline .gitignore]: We might need to be careful about ignoring this folder, particularly if/when we start using npm modules.

File: core/server/api.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Global API configuration
 API = {};
 
+// Swagger UI tags
+OrganizationTag = "Organizations";
```
Comment (brylie) [inline core/server/api.js]: We probably don't need these to be global variables. Can they be attached to the API object? E.g. `API.tags.organizationTag`

File: departments/server/api.js [hunk 3]
```diff
@@ -2,18 +2,99 @@
...
 API.v1.addCollection(Departments, {
   routeOptions: {
-    authRequired: true
+    authRequired: false
+  },
+  endpoints: {
+    get: {
+      swagger: {
+        tags: [
+          DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 4]
```diff
@@ -2,18 +2,99 @@
...
+          "200": {
+            description: "One department"
+          }
+        }
+      }
+    },
+    post: {
+      swagger: {
+        tags: [
+          DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 5]
```diff
@@ -2,18 +2,99 @@
...
+          "200": {
+            description: "Return department that was added."
+          }
+        }
+      }
+    },
+    put: {
+      swagger: {
+        tags: [
+          DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 6]
```diff
@@ -2,18 +2,99 @@
...
+          "200": {
+            description: "Returns updated department."
+          }
+        }
+      }
+    },
+    delete: {
+      swagger: {
+        tags: [
+          DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 7]
```diff
@@ -2,18 +2,99 @@
...
+          "200": {
+            description: "Successful delete."
+          }
+        }
+      }
+    },
+    getAll: {
+      swagger: {
+        tags: [
+          DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 8]
```diff
@@ -2,18 +2,99 @@
...
-    // Get organizationId from URL parameters
-    const organizationId = this.urlParams.id;
-    // Fetch all departments of the given organization
-    const departments = Departments.find({ "organizationId": organizationId }).fetch();
+API.v1.addRoute('organizations/:id/departments', {authRequired: false}, {
+  get: {
+    swagger: {
+      tags: [
+        OrganizationTag,
+        DepartmentTag
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.department`

File: departments/server/api.js [hunk 9]
```diff
@@ -2,18 +2,99 @@
...
-  get: function () {
-    // Get organizationId from URL parameters
-    const organizationId = this.urlParams.id;
-    // Fetch all departments of the given organization
-    const departments = Departments.find({ "organizationId": organizationId }).fetch();
+API.v1.addRoute('organizations/:id/departments', {authRequired: false}, {
+  get: {
+    swagger: {
+      tags: [
+        OrganizationTag,
```
Comment (brylie) [inline departments/server/api.js]: Update this to `API.tags.organization`

**Comment by brylie** [comment]
When trying to run the latest version of this code, I get the following error:

```
=> Errors prevented startup:

   While selecting package versions:
   error: Potentially incompatible change required to top-level dependency: apinf:restivus-swagger 0.1.1, was 0.1.2.
   Constraints on package "apinf:restivus-swagger":

   To allow potentially incompatible changes to top-level dependencies, you must pass --allow-incompatible-update on the command line.
```

When I run `meteor --allow-incompatible-update`, the `apinf:restivus-swagger` package is *downgraded* to 0.1.1.

**Comment by brylie** [comment]
It looks like 0.1.2 has not been published to Atmosphere:

https://atmospherejs.com/apinf/restivus-swagger

File: organizations/server/api.js [hunk 10]
```diff
@@ -2,6 +2,73 @@
...
 API.v1.addCollection(Organizations, {
   routeOptions: {
-    authRequired: true
+    authRequired: false
+  },
+  endpoints: {
+    get: {
+      swagger: {
+        tags: [
+          OrganizationTag
```
Comment (brylie) [inline organizations/server/api.js]: Update this to `API.tags.organization`

File: organizations/server/api.js [hunk 11]
```diff
@@ -2,6 +2,73 @@
...
+          "200": {
+            description: "One organization"
+          }
+        }
+      }
+    },
+    post: {
+      swagger: {
+        tags: [
+          OrganizationTag
```
Comment (brylie) [inline organizations/server/api.js]: Update this to `API.tags.organization`

File: organizations/server/api.js [hunk 12]
```diff
@@ -2,6 +2,73 @@
...
+          "200": {
+            description: "Return organization that was added."
+          }
+        }
+      }
+    },
+    put: {
+      swagger: {
+        tags: [
+          OrganizationTag
```
Comment (brylie) [inline organizations/server/api.js]: Update this to `API.tags.organization`

File: organizations/server/api.js [hunk 13]
```diff
@@ -2,6 +2,73 @@
...
+          "200": {
+            description: "Returns updated organization."
+          }
+        }
+      }
+    },
+    delete: {
+      swagger: {
+        tags: [
+          OrganizationTag
```
Comment (brylie) [inline organizations/server/api.js]: Update this to `API.tags.organization`

File: organizations/server/api.js [hunk 14]
```diff
@@ -2,6 +2,73 @@
...
+          "200": {
+            description: "Successful delete."
+          }
+        }
+      }
+    },
+    getAll: {
+      swagger: {
+        tags: [
+          OrganizationTag
```
Comment (brylie) [inline organizations/server/api.js]: Update this to `API.tags.organization`

**Comment by jykae** [comment]
Changes made, and published new release of restivus-swagger to Atmosphere.

**Pull request merged**